feat: adding an error_reason when apy calculation fails for a vault

mariuspod opened the pull request:
Related issue # (if applicable):

### What I did:

### How I did it:

### How to verify it:

### Checklist:
- [ ] I have tested it locally and it works
- [ ] I have included any relevant documentation for the repo maintainers
- [ ] (If fixing a bug) I have added a test to the test suite to test for this particular bug

#### Adding a Network
If the purpose of your PR is to add support for a new network, please copy the checklist from [here](https://github.com/yearn/yearn-exporter/blob/master/.github/new_network.md) into this PR conversation, and use it to track your changes.

DarkGhost7 reviewed (approved):
tested locally working well

mariuspod commented:
@DarkGhost7 thanks for testing this!
I've added the `error_reason` to the apy previews as well and fixed a bug where the message would be added to the wrong key inside the Apy object.

DarkGhost7 reviewed (approved):
lgtm
